### ISSUE TITLE
Wrong null pointer inspection triggered

### ISSUE THREAD

**Issue by Koc** [kalessil/phpinspectionsea#545]
this is a class method with typehint `public function ololo(City $city = null)` and we have check for variable value

![default](https://user-images.githubusercontent.com/191082/30641101-7b9bcfb0-9e0e-11e7-836a-b057d3adca83.png)

**Comment by rentalhost** [comment]
Thanks for reporting. 
What is the type of `$city` identified by PS? (hold CTRL and point to this variable).

**Comment by Koc** [comment]
![default](https://user-images.githubusercontent.com/191082/30641667-aff42b84-9e10-11e7-9dd6-23927f864065.png)

**Comment by rentalhost** [comment]
I could reproduce it with:

```php
class User { public function okay(): void { } }

function test(User $user = null)
{
    if (false) {
        // It applied here it works fine.
    }
    elseif ($user) {
        // But here not.
        $user->okay();
    }
}

```

Seems that only `if()` and `else if()` is validated, and `elseif()` is not.

**Comment by Koc** [comment]
thank you for problem localization

**Comment by kalessil** [comment]
Oh, a new bug related to NPEs. Today is NPEs inspection bug day =)

**Comment by kalessil** [comment]
This fixed, thank you @Koc for reporting and @rentalhost for precise diagnostics!

**Comment by Koc** [comment]
@kalessil  @rentalhost  Issue returns with last update

![default](https://user-images.githubusercontent.com/191082/32184201-311e25ae-bda4-11e7-97b6-d5d5b05cf5c1.png)

**Comment by kalessil** [comment]
@Koc : do you mean Php Inspections (EA Extended) v2.3.11.1 update?

**Comment by Koc** [comment]
yes, `Version: 2.3.11.1`,

```
PhpStorm 2017.3 EAP
Build #PS-173.3415.21, built on October 25, 2017
PhpStorm EAP User
Expiration date: November 24, 2017
JRE: 1.8.0_152-release-1024-b6 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0
```

**Comment by kalessil** [comment]
@Koc : congrats, you discovered a new case - reproduced and fixed
